### DISCUSSION TITLE
does 4channel ESP8266 / ESP-01 relay module  ... work with esp-homekit  ??

### DISCUSSION THREAD

**Discussion by lickonline** [RavenSystem/esp-homekit-devices#1571] [Q&A]
Hi,
does anyone know, whether a **4ch** ESP8266 / ESP-01 relay module 12V from  LC technology  does work with esp-homekit  ??
I could flash the 4ch module with haa 7.4.0, I got 4 switches in homekit, but relais doesn't work.
I read something about "serial send command", works this relay module not with gpio's?
any way to get this 4ch module to work with haa ??

___________________________

(if not, any suggestions for alternative 3ch or 4ch modules (as small as possible) for esp-homekit/haa are welcome!

**Comment by LachNed** [comment]
Hello,
I also want to use a LC technology ESP-01 works with HomeKit.
So I got to it work with Tasmota by using the code and rules from https://templates.blakadder.com/LC-ESP8266.html.
I guess that it also should work with HAA...
Then I played around with the serial send codes from the Tasmota site and got it  running in HAA

Here's my JSON (I'm sure it's not totally correct and it could be optimized)

{
  "c":{
    "l":13,
    "b":[{"g":0,"t":5}],
    "r":[{"n":0, "s":9600, "p":0, "b":0}]
        },
  "a": [{
    "t": 1,
    "0": { "u":[{"n":0, "v": "A00100A1" }]},
    "1": { "u":[{"n":0, "d":100, "v": "A00101A2" }]}
         }]
}

To all the other experts... Please help us with hints how to use it with the 2CH or the 4CH-variants

Thanks

**Comment by vglucas** [comment]
Yes, it is possible to install haa
This is the json for board 4ch

{"c":{"r":[{"n":0}],"0":{"u":[{"n":0,"v":"41542b5253540d0a"}]}},
"a":[
{"0":{"u":[{"n":0,"v":"a00100a1"}]},"1":{"u":[{"n":0,"v":"a00101a2"}]}},
{"0":{"u":[{"n":0,"v":"a00200a2"}]},"1":{"u":[{"n":0,"v":"a00201a3"}]}},
{"0":{"u":[{"n":0,"v":"a00300a3"}]},"1":{"u":[{"n":0,"v":"a00301a4"}]}},
{"0":{"u":[{"n":0,"v":"a00400a4"}]},"1":{"u":[{"n":0,"v":"a00401a5"}]}}
]}

**Reply by LachNed** [reply]
Perfect...

thank you...
Now I can see, that only a copy and modification (as I created my code) is now the right way....

Can somebody explain, how to get such a JSON by coding and not by copying and changing code?
How should I start, if I have a new relay module or switch?

**Reply by LachNed** [reply]
Wrong Text -I meant... "is NOT the right way"

**Reply by vglucas** [reply]
@LachNed 
what's the problem??

**Reply by LachNed** [reply]
my problem is...
I just copied the 2 lines from another JSON and took the "A00100A1" from an example of a Tasmota-config...

But I would like to know how to write such JSON code... without copying different code fragments from different examples...
Is there a tutorial to get these UART-codes?

**Reply by vglucas** [reply]
view this page

https://github.com/RavenSystem/esp-homekit-devices/wiki/HAA-Templates

**Reply by LachNed** [reply]
Thank you... I know that page but I don't know how to find the special commands like "A00100A1"
Anyway... I will try in future to learn more and more that HAA-JSON codes
But I think a website with detailed step by step tutorials would help a lot of people like me...

Thank you and thanks to all who bring HAA forward
